### ISSUE TITLE
Plone 5.1 delivers improper search results page when enabled from search box.

### ISSUE THREAD

**Issue by zopyx** [plone/Products.CMFPlone#2697]
Goto 

https://www.plone-demo.info/documents-all-content

and check "only in current section" and search for "document" and click the "Search" button.

This opens a URL 

https://www.plone-demo.info/@@search?path=%2Fplone%2Fdocuments-all-content&SearchableText=document

and shows 10 search results.

![screenshot-www plone-demo info-2019-01-16-11-50-40](https://user-images.githubusercontent.com/594239/51244289-01059a80-1985-11e9-9ae2-fa9ab5c3bb4f.png)

When you click on "Relevance" or a different tab then Plolne displays 21 search items in 3 batches.
Even showing a batch navigation with 3 batches for 21 items with a batch size of 10 is just broken.

![screenshot-www plone-demo info-2019-01-16-11-51-20](https://user-images.githubusercontent.com/594239/51244383-432edc00-1985-11e9-95d3-d0d116223f3b.png)

**Comment by tmassman** [comment]
Without JS the links work fine (path is preserved). With JS the path parameter is discarded.

A fix could be to add the missing `path` to the template:
```
        <input type="hidden" name="sort_on" value="" />
        <input type="hidden" name="sort_order" value="" />
        <input type="hidden" name="path" value="" tal:attributes="value request/form/path | nothing" />
```

The problem is that the UI does not give any feedback about the path restriction. There should be an option for the user to see the current path selection and disable it, if the user wants to.

It is also confusing that the path restriction is still shown on the search viewlet in the header, but the value changes to the Plone site id (the current path).

**Comment by sureshvv** [comment]
Would like to help. Have you identified what needs to be done?

**Comment by tmassman** [comment]
Thank you @sureshvv!

We need:
- the form parameter passed to the search results (input element with name `path`)
- a visual representation of the current `path` selected (e.g. breadcrumb style)
- an option for the user to disable the path selection

I could imagine a checkbox with the label of the current path selected (whatever style we choose to use). When the user unchecks the checkbox, the search scope will be global.

We also have a problem with the search box in the header on the search page. It shows the current searched text (which is fine) but also the checkbox **only in current section**, but with the wrong value (the id of the Plone site in this case, because the context of the `@@search` page is the Plone site). This can conflict with the input element we need to pass the current path selection (having 2 elements with the same `name` attribute but different values). We should try to either hide this checkbox on the search page, or the complete search box (I would prefer the first option).

**Comment by sureshvv** [comment]
Thank you @tmassman 

Setting up dev env now. Will update you.

**Comment by sureshvv** [comment]
Should we make @@search sensitive to context, i.e. /plone/documents-all-content/@@search will be url?

**Comment by tmassman** [comment]
> Should we make @@search sensitive to context, i.e. /plone/documents-all-content/@@search will be url?

I thought about that too. With that we should take a look at the default `robots.txt` settings, which only block the `@@search` view at the root level.
```
Disallow: /@@search
```
It would indeed solve the problem with the local path in the search viewlet.

**Comment by zopyx** [comment]
I second the approach for making `@@search` sensitive to the current context.

**Comment by sureshvv** [comment]
I worked off the 5.1.x branch. Not sure where I should file the pull request

**Comment by sureshvv** [comment]
Created Pull Request: https://github.com/plone/Products.CMFPlone/pull/2710

**Comment by jensens** [comment]
I close the issue, because it addresses a Plone version that is no longer supported. 
If you think this is wrong please reopen the issue and assign a matching milestone.
